primitives/sr-primitives/src/lib.rs

```diff
@@ -348,56 +348,41 @@ impl From<ed25519::Signature> for AnySignature {
 
 #[derive(Eq, PartialEq, Clone, Copy, Decode, Encode, RuntimeDebug)]
 #[cfg_attr(feature = "std", derive(Serialize))]
-/// Reason why an extrinsic couldn't be applied (i.e. invalid extrinsic).
-pub enum ApplyError {
-	/// General error to do with the permissions of the sender.
-	NoPermission,
-
-	/// General error to do with the state of the system in general.
-	BadState,
-
+/// Reason why an extrinsic couldn't be included into a block.
+pub enum InclusionError {
```

Comment:
Since there is only one variant left I wonder if the existence of this type is justified. I think it would be way clearer to return `TransactionValidityError` instead of a wrapper type.

Reply:
So yeah, I actually left a comment on it somewhere (in the commit msg, I guess? : P) but now you say this I think I see no reason to keep it this way.

Let's do it. However, would you be OK with the fact that we would return `TransactionValidityError` from `apply_extrinsic` (that you [stated here](https://github.com/paritytech/substrate/pull/4134/files#r347827672))?